general:
  world_frame: "odom"
  logging_level: debug
  allow_reset_map_service: true

map:
  type: wavelet_octree
  min_cell_width: { meters: 0.2 }

map_operations:
  - type: threshold_map
    once_every: { seconds: 2.0 }
  - type: prune_map
    once_every: { seconds: 10.0 }
  - type: publish_map
    once_every: { seconds: 10.0 }

measurement_integrators:
  ouster_os1:
    projection_model:
      type: ouster_projector
      lidar_origin_to_beam_origin: { millimeters: 27.67 }
      lidar_origin_to_sensor_origin_z_offset: { millimeters: 36.18 }
      elevation:
        num_cells: 64
        min_angle: { degrees: -15.594 }
        max_angle: { degrees: 17.743 }
      azimuth:
        num_cells: 1024
        min_angle: { degrees: -180.0 }
        max_angle: { degrees: 180.0 }
    measurement_model:
      type: continuous_beam
      angle_sigma: { degrees: 0.05 }
      range_sigma: { meters: 0.05 }
      scaling_free: 0.2
      scaling_occupied: 0.4
    integration_method:
      type: wavelet_integrator
      min_range: { meters: 1.0 }
      max_range: { meters: 20.0 }

inputs:
  - type: pointcloud
    topic_name: "/os1_cloud_node/points"
    topic_type: ouster
    measurement_integrator_names: ouster_os1
    undistort_motion: true
    topic_queue_length: 10
    max_wait_for_pose: { seconds: 1.0 }
